string file_name
float32 alpha
bool is_url
